lib/rules/prefer-const.js

```diff
@@ -203,6 +203,23 @@ function groupByDestructuring(variables, ignoreReadBeforeAssign) {
     return identifierMap;
 }
 
+/**
+ * Finds the nearest parent of node with a given type.
+ *
+ * @param {ASTNode} node – The node to search from.
+ * @param {string} type – The type field of the parent node.
+ * @returns {ASTNode} The closest ancestor with the specified type; null if no such ancestor exists.
+ */
+function findUp(node, type) {
+    if (!node) {
+        return null;
+    }
+    if (node.type === type) {
+        return node;
+    }
+    return findUp(node.parent, type);
```

Comment:
I tend to prefer recursion as a matter of style, but I'm happy to do this iteratively. If you feel strongly, I'll change it.